fuzz/fuzz_targets/c_rust_merkle.rs

```diff
@@ -46,11 +46,11 @@ fn do_test(data: &[u8]) {
             }
             println!("{}", rust_data.cmr());
             println!("{}", rust_data.amr());
-            println!("{}", rust_data.imr());
+            println!("{}", rust_data.ihr());
```

Comment:
e4522021dc59d682901a5a5a2982a65c4ba7d4b1: @apoelstra @roconnor-blockstream Are IMRs entirely replaced by IHRs? I thought, some use cases are called IMR and some are called IHR, in which case a blind find-and-replace also renames use cases where IMR is still the valid term.

Reply:
Grepping the Simplicity codebase for `IMR` and `Identity Merkle Root` shows some scattered remaining references (e.g. in the tech report it appears no renaming has been done) but nothing that looks like a principled distinction.

I believe the literal text `imr_` stays in the preimage of the IV for IHR computation for witness and disconnect nodes, but this codebase never sees the preimage text. We just use precomputed IVs that we get from the C code.